FutureNorthants/northants-website
# Frontend homepage promo banner expansion

### Setup
This PR relies upon two others so there's a bit of work to do to get it all up and running:

**Backend**
PR: https://github.com/FutureNorthants/northants-website/pull/190
This provides the graphql schema update and associated query functionality that will provide expanded promo banner data. Once checked out, `make stop` and `make start` your backend containers. You may need to do a `drush cim` too.

**Design System**
PR: https://github.com/FutureNorthants/northants-design-system/pull/78
This provides the new PromoBlock structure that the front end will use. Once checked out, you'll need to locally publish a copy for your frontend instance to use with `yalc publish`.

**Front End**
Stop your local instance if it's running. Check out this branch. Pull in the local copy of the design system with `yalc add northants-design-system` and then run `yarn install --skip-integrity-check` (to pull in the sanitize-html module). Now run `yarn dev` to start the front end server. Open a new terminal window and go back into your frontend folder. Run `yarn codegen`, it should run OK (if it doesn't, talk to me right away). If you run `git status` afterwards there shouldn't be any changes in __generated folders (you can ignore the package file changes it'll show) - if there any changes let me know right away again.

You should be now running the new front end changes with the new promo block design system structure in place.

### Testing
1. Visit the homepage. Should work as normal, but the unitary map bit should now be below the news stories. The existing promo banner should be unchanged. If you have previously tested the backend PR you'll see any new promo bits in the block below the promo banner. If you have not prevously tested the backend PR then...
2. Edit the homepage in the CMS and go to the news & promotions tab. Try clearing out the existing one(s). Verify nothing show on homepage and no problems. Add one promo banner entry and verify it appears OK on the homepage. Add more, verifying each time, and verify you can display up to 4 in total (1 banner and 3 tiles).

### Deployment Notes
We'll need to publish a design system update first and add another commit to this PR to update the package files correctly (_not_ using the local design system copy!). Then, deploy backend PR and this one.
